cogs/send_introduction_reminders.py

```diff
@@ -134,22 +134,29 @@ async def send_introduction_reminders(self) -> None:
                     if message_contains_opt_in_out_button:
                         await message.edit(view=None)
 
-                await member.send(
-                    content=(
-                        "Hey! It seems like you joined "
-                        f"the {self.bot.group_short_name} Discord server "
-                        "but have not yet introduced yourself.\n"
-                        "You will only get access to the rest of the server after sending "
-                        "an introduction message."
-                    ),
-                    view=(
-                        self.OptOutIntroductionRemindersView(self.bot)
-                        if settings["SEND_INTRODUCTION_REMINDERS"] == "interval"
-                        else None  # type: ignore[arg-type]
-                    ),
-                )
+                if member in guild.members:
+                    await member.send(
+                        content=(
+                            "Hey! It seems like you joined "
+                            f"the {self.bot.group_short_name} Discord server "
+                            "but have not yet introduced yourself.\n"
+                            "You will only get access to the rest of the server after sending "
+                            "an introduction message."
+                        ),
+                        view=(
+                            self.OptOutIntroductionRemindersView(self.bot)
+                            if settings["SEND_INTRODUCTION_REMINDERS"] == "interval"
+                            else None  # type: ignore[arg-type]
+                        ),
+                    )
 
-                await SentOneOffIntroductionReminderMember.objects.acreate(member_id=member.id)
+                    await SentOneOffIntroductionReminderMember.objects.acreate(member_id=member.id)
+                else:
+                    logger.info(
+                        "Member: {member} left the server before the introduction reminder could be sent."
+                    )
+                    member_needs_reminder = False
```

Comment:
This value does not need to be edited as it was used as a flag above to enter this block.

https://github.com/CSSUoB/TeX-Bot-Py-V2/blob/51d18c6b471b72387d689f0ea5ad890c1127f9a8/cogs/send_introduction_reminders.py#L125